sentinel-adapter/sentinel-spring-cloud-gateway-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/sc/SentinelGatewayFilter.java

```diff
@@ -47,16 +47,24 @@
 
     private final int order;
 
+    private final GatewayParamParser<ServerWebExchange> paramParser;
+
     public SentinelGatewayFilter() {
         this(Ordered.HIGHEST_PRECEDENCE);
     }
 
     public SentinelGatewayFilter(int order) {
-        this.order = order;
+        this(order, new ServerWebExchangeItemParser());
     }
 
-    private final GatewayParamParser<ServerWebExchange> paramParser = new GatewayParamParser<>(
-        new ServerWebExchangeItemParser());
+    public SentinelGatewayFilter(RequestItemParser<ServerWebExchange> serverWebExchangeItemParser) {
+        this(Ordered.HIGHEST_PRECEDENCE, serverWebExchangeItemParser);
+    }
+
+    public SentinelGatewayFilter(int order, RequestItemParser<ServerWebExchange> serverWebExchangeItemParser) {
+        this.order = order;
+        this.paramParser = new GatewayParamParser<>(serverWebExchangeItemParser == null ? new ServerWebExchangeItemParser() : serverWebExchangeItemParser);
```

Comment:
if our requestItemParser is can not be null, I think assert it cannot null is better. because we have provided the default value constructor.